flake.nix

```diff
@@ -1,72 +1,154 @@
 # SPDX-FileCopyrightText: 2024 - 2025 Mass Labs
 #
 # SPDX-License-Identifier: MIT
-
 {
   description = "Mass Market Network Schema";
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-24.11";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
     utils.url = "github:numtide/flake-utils";
     gomod2nix = {
-      url = "github:tweag/gomod2nix";
+      # url = "github:tweag/gomod2nix";
+      # fix builds with go 1.23
+      # https://github.com/nix-community/gomod2nix/pull/168
+      url = "github:obreitwi/gomod2nix/fix/go_mod_vendor";
       inputs.nixpkgs.follows = "nixpkgs";
     };
+    bryce-mmrs-py = {
+      url = "github:robinbryce/draft-bryce-cose-merkle-mountain-range-proofs";
+      flake = false;
+    };
   };
 
   outputs = {
     nixpkgs,
     utils,
     gomod2nix,
+    bryce-mmrs-py,
     ...
   }:
     utils.lib.eachDefaultSystem (system: let
       pkgs = import nixpkgs {
         inherit system;
-        overlays = [ gomod2nix.overlays.default ];
+        overlays = [gomod2nix.overlays.default];
+      };
+
+      ourPython = pkgs.python312;
+
+      # https://github.com/ethereum/eth-typing/issues/63#issuecomment-2291678106
+      pyunormalize = ourPython.pkgs.buildPythonPackage rec {
+        pname = "pyunormalize";
+        version = "15.1.0";
+        src = ourPython.pkgs.fetchPypi {
+          inherit pname version;
+          hash = "sha256-z0qHRRoPHLdpEaqX9DL0V54fVkovDITOSIxzpzkBtsE=";
+        };
+        buildInputs = [pinnedPython.pkgs.setuptools];
+        # le sigh...
+        postUnpack = ''
+          sed -i 's/version=get_version()/version="${version}"/' ${pname}-${version}/setup.py
+          sed -i 's/del _version//' ${pname}-${version}/${pname}/__init__.py
+        '';
       };
 
-      # web3 needs parsimonious v0.9.0
-      # https://github.com/ethereum/web3.py/issues/3110#issuecomment-1737826910
       packageOverrides = self: super: {
         parsimonious = super.parsimonious.overridePythonAttrs (old: rec {
           pname = "parsimonious";
-          version = "0.9.0";
-          src = pkgs.python3.pkgs.fetchPypi {
+          version = "0.10.0";
+          src = self.fetchPypi {
+            inherit pname version;
+            sha256 = "sha256-goFgDaGA7IrjVCekq097gr/sHj0eUvgMtg6oK5USUBw=";
+          };
+        });
+
+        asn1tools = super.asn1tools.overridePythonAttrs (old: rec {
+          pname = "asn1tools";
+          version = old.version;
+          src = old.src;
+          doCheck = false; # TODO: just on darwin..?
+        });
+
+        eth-keys = super.eth-keys.overridePythonAttrs (old: rec {
+          pname = "eth-keys";
+          version = "0.6.1";
+          # for some reason the pypi version gives me 404...?
+          # src = self.fetchPypi {
+          #   inherit pname version;
+          #   sha256 = "";
+          # };
+          src = ./eth_keys-0.6.1.tar.gz;
+          doCheck = false;
+        });
```

Comment:
eth-keys is already in the package at `python312Packages.eth-keys` it is 0.6.0 though.